drivers/flash/spi_flash.c

```diff
@@ -0,0 +1,523 @@
+	u32_t support_dtr : 1;
+	u32_t support_1_2_2_fast_read : 1;
+	u32_t support_1_4_4_fast_read : 1;
+	u32_t support_1_1_4_fast_read : 1;
+	u32_t unused2 : 9;
+
+	u32_t density;
+
+	struct {
+		u8_t dummy_clocks : 4;
+		u8_t mode_clocks : 4;
+		u8_t opcode;
+	} read_opcodes[4]; /* 1-4-4, 1-1-4, 1-1-2, 1-2-2 */
+
+	u32_t support_2_2_2_fast_read : 1;
+	u32_t unused3 : 3;
+	u32_t support_4_4_4_fast_read : 1;
+	u32_t unused4 : 27;
+
+	struct {
+		u8_t dummy_clocks : 4;
+		u8_t mode_clocks : 4;
+		u8_t opcode;
+	} read_opcodes_2[4]; /* reserved, 2-2-2, reserved, 4-4-4 */
+
+	struct {
+		u8_t size;
+		u8_t erase_opcode;
+	} sector_sizes[4];
+
+	struct {
+		u8_t mult : 4, time : 4;
+	} sectore_erase_times[4];
+
+	u8_t page_mult : 4;
+	u8_t page_size_shift : 4;
+	u8_t page_program_time : 4;
+	u8_t first_byte_program_time : 4;
+	u8_t additional_byte_program_time : 4;
+	u8_t chip_erase_time : 7;
+	u8_t unused5 : 1;
+
+	u32_t suspend_info;
+
+	u8_t program_resume_opcode;
+	u8_t program_suspend_opcode;
+	u8_t resume_opcode;
+	u8_t suspend_opcode;
+
+	u32_t power_down_info;
+};
+
+#define SFDP_BFPT_ID		0xff00  /* Basic Flash Parameter Table */
+#define SFDP_SECTOR_MAP_ID	0xff81  /* Sector Map Table */
+
+struct spi_flash_data {
+	struct spi_config config;
+	struct k_sem sem;
+
+	u32_t flash_size, page_size;
+	u8_t addr_len;
+	u8_t read_opcode, read_dummy;
+	u8_t write_opcode, write_dummy;
+	u8_t global_unprotect_opcode;
+
+	union {
+		u32_t bfpt[16];
+		struct sfdp_basic_flash_parameters bfp;
+	};
+	u32_t sector_map[16];
+};
+
+#define CMD_IDENTIFY		0x9F
+#define CMD_DISCOVER_PARAMETERS	0x5A
+#define CMD_READ_STATUS		0x05
+#define CMD_READ_MEMORY		0x03
+#define CMD_READ_CONFIG		0x35
+#define CMD_WRITE_STATUS_CONFIG	0x01
+#define CMD_PROGRAM_PAGE	0x02
+#define CMD_WRITE_ENABLE	0x06
+#define CMD_WRITE_DISABLE	0x04
+
+#define STATUS_BUSY		0x01
+
+#define MAX_SFDP_HEADERS	4
+
+static inline int spi_flash_reg_read(struct device *dev, u8_t cmd_id)
+{
+	struct spi_flash_data *const data = dev->driver_data;
+	u8_t buf[2];
+	struct spi_buf cmd = {
+		.buf = buf,
+		.len = sizeof buf,
+	};
+
+	buf[0] = cmd_id;
+	if (spi_transceive(&data->config, &cmd, 1, &cmd, 1) != 0) {
+		return -EIO;
+	}
+
+	return buf[1];
+}
+
+static inline void wait_for_flash_idle(struct device *dev)
+{
+	while (spi_flash_reg_read(dev, CMD_READ_STATUS) & STATUS_BUSY)
+		;
+}
+
+static inline int spi_flash_write_sr_cr(struct device *dev, u8_t sr, u8_t cr)
+{
+	struct spi_flash_data *const data = dev->driver_data;
+	u8_t buf[3];
+	struct spi_buf cmd = {
+		.buf = buf,
+		.len = sizeof buf,
+	};
+
+	buf[0] = CMD_WRITE_STATUS_CONFIG;
+	buf[1] = sr;
+	buf[2] = cr;
+	if (spi_transceive(&data->config, &cmd, 1, &cmd, 1) != 0) {
+		return -EIO;
+	}
+	wait_for_flash_idle(dev);
+	return 0;
+}
+
+static inline int spi_flash_id(struct device *dev)
+{
+	struct spi_flash_data *const data = dev->driver_data;
+	uint8_t buf[4] = { CMD_IDENTIFY };
+	struct spi_buf bufs = { .buf = &buf, sizeof buf };
```

Comment:
always 1 line per attribute assignement. (you are actually doing it above, line 160, properly). 

Apply that everywhere.